docs/code-style-guide.md

```diff
@@ -18,6 +19,45 @@ Our C++ coding style is based off of [Google's C++ Style Guide](https://google.g
 If you want to know more about our coding style you can take a look at our [clang-format configuration file](https://github.com/UBC-Thunderbots/Software/blob/master/.clang-format).
 
 
+### C-Specific Stuff
```

Comment:
über mega nit: Maybe `C-Specific Guidelines`?, I doesn't really matter so its cool if u don't change it either

Reply:
changed